extensions/tasks-parent/tasks-client/src/test/java/io/resys/thena/tasks/tests/TaskHistoryTest.java

```diff
@@ -0,0 +1,103 @@
+import io.resys.thena.tasks.client.api.model.ImmutableChangeTaskPriority;
+import io.resys.thena.tasks.client.api.model.ImmutableChangeTaskStatus;
+import io.resys.thena.tasks.client.api.model.ImmutableCreateTask;
+import io.resys.thena.tasks.client.api.model.Task;
+import io.resys.thena.tasks.tests.config.TaskPgProfile;
+import io.resys.thena.tasks.tests.config.TaskTestCase;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONException;
+import org.junit.jupiter.api.Assertions;
+
+import java.time.Duration;
+
+@Slf4j
+@QuarkusTest
+@TestProfile(TaskPgProfile.class)
+public class TaskHistoryTest extends TaskTestCase {
+
+  private static final String repoName = TaskUpdateTest.class.getSimpleName();
+
+  @org.junit.jupiter.api.Test
+  public void createTaskAndUpdateAndGetHistory() throws JsonProcessingException, JSONException {
+    final var client = getClient().repo().query().repoName(repoName).createIfNot().await().atMost(atMost);
+
+    Task createdTask_1 = client.tasks().createTask().createOne(ImmutableCreateTask.builder()
+            .targetDate(getTargetDate())
+            .title("very important title no: init")
+            .description("first task ever no: init")
+            .priority(Task.Priority.LOW)
+            .addRoles("admin-users", "view-only-users")
+            .userId("user-1")
+            .reporterId("reporter-1")
+            .build())
+        .await().atMost(atMost);
+
+    Task createdTask_2 = client.tasks().createTask().createOne(ImmutableCreateTask.builder()
+            .targetDate(getTargetDate())
+            .title("very important title no: init")
+            .description("second task ever no: init")
+            .priority(Task.Priority.LOW)
+            .addRoles("admin-users", "view-only-users")
+            .userId("user-1")
+            .reporterId("reporter-1")
+            .build())
+        .await().atMost(atMost);
+
+
+    client.tasks().updateTask().updateOne(ImmutableChangeTaskStatus.builder()
+            .userId("tester-bob")
+            .taskId(createdTask_1.getId())
+            .targetDate(getTargetDate().plusDays(1).plusHours(1))
+            .status(Task.Status.IN_PROGRESS)
+            .build())
+        .await().atMost(atMost);
+
+    client.tasks().updateTask().updateOne(ImmutableChangeTaskPriority.builder()
+            .userId("tester-bob")
+            .taskId(createdTask_1.getId())
+            .targetDate(getTargetDate().plusDays(1).plusHours(1))
+            .priority(Task.Priority.HIGH)
+            .build())
+        .await().atMost(atMost);
+
+    client.tasks().updateTask().updateOne(ImmutableAssignTaskReporter.builder()
+            .userId("tester-bob")
+            .taskId(createdTask_1.getId())
+            .targetDate(getTargetDate().plusDays(1).plusHours(1))
+            .reporterId("citizen jane")
+            .build())
+        .await().atMost(atMost);
+
+    final var history = client.tasks().queryTaskHistory().get("1_TASK").await().atMost(Duration.ofMinutes(1));
```

Comment:
When pushing something to debug, in our case, this is a huge object.  It would be nicer to read if you prettyprint it, using `JsonObject.mapFrom(history).encodePrettily();`

Also, remember that there is application.properties in test/resources that controls what levels and what to log.